conjur/wrapper/keystore_adapter.py

```diff
@@ -31,52 +32,70 @@ def set_password(cls, identifier, key, val):
         """
         Method for setting a password in keyring
         """
-        keyring.set_password(identifier, key, val)
+        try:
+            keyring.set_password(identifier, key, val)
+        except keyring.errors.PasswordSetError as password_error:
+            raise KeyringAdapterSetError(f"unable to set key: {key} for identifier: "
+                                  f"{identifier}") from password_error
+        except keyring.errors.KeyringError as keyring_error:
+            raise KeyringAdapterGeneralError(f"unable to set key: {key} for identifier: "
+                                      f"{identifier}") from keyring_error
 
     @classmethod
     def get_password(cls, identifier, key):
         """
         Method for getting a password in keyring
         """
-        return keyring.get_password(identifier, key)
+        try:
+            return keyring.get_password(identifier, key)
+        except keyring.errors.KeyringError as keyring_error:
+            raise KeyringAdapterGeneralError(f"unable to get value for key: {key} "
+                                      f"for identifier: {identifier}") from keyring_error
 
-    # TODO add simple delete for one attribute only here in the adapter
     # pylint: disable=try-except-raise
     @classmethod
-    def delete_password(cls, identifier):
+    def delete_password(cls, identifier, key):
         """
         Method for deleting a password in keyring
         """
         try:
-            for attr in KEYSTORE_ATTRIBUTES:
-                keyring.delete_password(identifier, attr)
-        # TODO do not raise keyring module errors
-        # Catches when credentials do not exist in the keyring. If the key does not exist,
-        # the user has already logged out
-        except keyring.errors.PasswordDeleteError:
-            return
-        except keyring.errors.KeyringError:
-            raise
+            keyring.delete_password(identifier, key)
+        except keyring.errors.PasswordDeleteError as password_error:
+            raise KeyringAdapterDeletionError(f"unable to delete key: {key} for identifier: "
+                                       f"{identifier}") from password_error
+        except keyring.errors.KeyringError as keyring_error:
+            raise KeyringAdapterGeneralError(f"unable to delete key: {key} for identifier:"
+                                      f" {identifier}") from keyring_error
 
     @classmethod
     def get_keyring_name(cls):
         """
         Method to get the system's keyring name
         """
-        return keyring.get_keyring().name
+        # keyring.get_keyring() can throw various types of exceptions
```

Comment:
# keyring.get_keyring() can throw various types of Exceptions.
        # Some are OS exceptions and that is the reason we catch general Exceptions.
        # Note that this is a critical path, if we get error here the entire app could 
        # not function

Reply:
why would the app not function? We would print to the NETRC no?

Reply:
@mbenita-Cyberark